### DISCUSSION TITLE
DashboardController

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#931] [Q&A (Help)] [answered]
This is probably not a problem related to Backpack but perhaps a shortcoming of mine which not even GPT was able to resolve..
In my web.php I have this code:

```
<?php

use Illuminate\Support\Facades\Route;
use App\Http\Controllers\DashboardController;


Route::get('/', function () {
    if (backpack_auth()->check()) {
        return redirect()->route('backpack.dashboard');
    } else {
        return view('welcome');
    }
});
```
Which works perfectly, but now I created a DashboardController because I need to run queries and pass data to dashboard.blade.php
How can I do ?
GPT suggests that I add the line to web.php
`Route::get('dashboard', [DashboardController::class, 'index'])->name('backpack.dashboard');`
but it doesn't work, due to an error..
Some advice ? Thanks for help :)

**Comment by jcastroa87** [comment, marked as answer]
Hi @blondie63 

What error did you get?

If you want to control dashboard routes yourself, in config/backpack/base.php set to false
```
'setup_dashboard_routes' => false,
``

Then in your web you can set routes like this:
```
Route::get('dashboard', 'AdminController@dashboard')->name('backpack.dashboard');
Route::get('/', 'AdminController@redirect')->name('backpack');
```

Let me know if you need something else.

Cheers.

**Reply by blondie63** [reply]
Thanks for your help !
At now i've decided to use a Livewire component to show my data in dashboard

**Comment by jcastroa87** [comment]
Great! i will close this issue then, but please feel free to re-open or create a new one if needed.

Cheers.
